corgidrp/detector.py

```diff
@@ -520,7 +525,155 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+	possible_sat_fwcs_arr = np.append((emgain_arr * fwcpp_arr)[:,np.newaxis], fwcem_arr[:,np.newaxis],axis=1)
+	sat_fwcs = sat_thresh * np.min(possible_sat_fwcs_arr,axis=1)
+
+	return sat_fwcs
+
+def flatfield_residuals(images,planet=None, N=3):
+    """Turn this dataset of image frames of neptune or uranus and create matched filters and estimate residuals after 
+     dividing from matched filters
+
+     Args:
+    	images (np.array): 3D Array of cropped neptune or uranus image frames
+    	planet (str): name of the planet uranus or neptune
+        N (int): number of images to be grouped. defaults to 3 for both Neptune and uranus. (If we use different number of dithers for neptune and uranus, option is provided to change N)
 
-    return sat_fwcs
+     Returns:
+    	matched_residuals (np.array): residual image frames of neptune or uranus divided by matched filter
+	"""
+    
+    raster_images = np.array(images)
+    images_split = np.array(np.split(np.array(raster_images),N))
+    matched_filters = np.array([np.nanmedian(np.stack(images_split[i],2),2) for i in np.arange(0,len(images_split))])
+    matched_filters_smooth = [gauss(matched_filters[i],3) for i in range(len(matched_filters))] 
+    
+    matched_residuals=[];
+    for j in range(len(raster_images)):
+        matched_residuals.append(raster_images[j]/matched_filters_smooth[int(np.floor(j//(len(raster_images)//len(matched_filters_smooth))))])
+    return(matched_residuals)
+	    
+def combine_flatfield_rasters(residual_images,cent=None,planet=None,band=None,im_size=420,rad_mask=None,  planet_rad=None, n_pix=165, n_pad=302):
+    """combine the dataset of residual image frames of neptune or uranus and create flat field 
+    	and associated error
+
+    	Args:
+        	residual_images (np.array): residual image frames of neptune and uranus 
+        	cent (np.array): centroid of the image frames
+        	planet (str):   name of the planet neptune or uranus
+        	band (str):  band of the observation band1 or band4
+            im_size (int): x-dimension of the planet image (in pixels= 420 for the HST images)
+            rad_mask (float): radius in pixels used for creating a mask for band (band 1 =1.25, band 4=1.75)
+            planet_rad (int): radius of the planet in pixels (planet_rad=54 for neptune, planet_rad=65)
+            n_pix (int): Number of pixels in radius covering the Roman CGI imaging FOV defaults to 165 pixels
+            n_pad (int): Number of pixels padded with '1s'  to generate the image size 1024X1024 pixels around imaging FOV (defaults to 302 pixels)
+
+        	
+    	Returns:
+        	full_qe (np.array): ideal flat field
+        	full_residuals (np.array): flat field created from uranus or neptune images
+        	std_resel (np.array):  Error of the flat field
+    """
+    n = im_size
+    full_qe=np.random.normal(1,.03,(n,n))
+    full_residuals = np.zeros((n,n))
+    if planet_rad==None:
```

Comment:
shouldn't this value be different for either Uranus or Neptune?

Reply:
yes, fixed it